gamemodes/terrortown/gamemode/server/sv_gamemsg.lua

```diff
@@ -233,48 +233,56 @@ end
 local cv_ttt_limit_spectator_chat = CreateConVar("ttt_limit_spectator_chat", "1", {FCVAR_ARCHIVE, FCVAR_NOTIFY})
 
 ---
--- Checks whether a @{Player} is able to listen to another @{Player}
+-- Returns whether or not the @{Player} can see the other @{Player}'s chat.
 -- @param string text
--- @param boolean team_only
--- @param Player listener
--- @param Player speaker
+-- @param boolean teamOnly
+-- @param Player reader
+-- @param Player sender
 -- @return boolean
 -- @hook
 -- @realm server
 -- @internal
-function GM:PlayerCanSeePlayersChat(text, team_only, listener, speaker)
-	if not IsValid(listener) then
+function GM:PlayerCanSeePlayersChat(text, teamOnly, reader, sender)
+	if not IsValid(reader) then
 		return false
 	end
 
-	if not IsValid(speaker) then
-		if IsEntity(speaker) then
-			return true
-		else
-			return false
-		end
+	if not IsValid(sender) and IsEntity(sender) then
+		return true
 	end
 
-	local sTeam = speaker:Team() == TEAM_SPEC
-	local lTeam = listener:Team() == TEAM_SPEC
+	local sTeam = sender:Team() == TEAM_SPEC
+	local lTeam = reader:Team() == TEAM_SPEC
 
 	if GetRoundState() ~= ROUND_ACTIVE -- Round isn't active
 	or not cv_ttt_limit_spectator_chat:GetBool() -- Spectators can chat freely
 	or not DetectiveMode() -- Mumbling
-	or not sTeam and (team_only and not speaker:IsSpecial() or not team_only) -- If someone alive talks (and not a special role in teamchat's case)
-	or not sTeam and team_only and (
-		speaker:IsInTeam(listener)
-		and not speaker:GetSubRoleData().unknownTeam
-		and not speaker:GetSubRoleData().disabledTeamChat
-		and not listener:GetSubRoleData().disabledTeamChatRecv
-		and hook.Run("TTT2AvoidTeamVoiceChat", speaker, listener) ~= false -- if the speaker and listener are in same team
-	) or sTeam and lTeam then -- If the speaker and listener are spectators
+	or not sTeam and (teamOnly and not sender:IsSpecial() or not teamOnly) -- If someone alive talks (and not a special role in teamchat's case)
+	or not sTeam and teamOnly and (
+		sender:IsInTeam(reader)
+		and not sender:GetSubRoleData().unknownTeam
+		and not sender:GetSubRoleData().disabledTeamChat
+		and not reader:GetSubRoleData().disabledTeamChatRecv
+		and hook.Run("TTT2CanSeeChat", reader, sender, teamOnly) ~= true
+	) or sTeam and lTeam then -- If the sender and reader are spectators
 		return true
 	end
 
 	return false
 end
 
+---
+-- Whether or not the @{Player} can receive the chat message.
+-- @param Player reader @{Player} who can receive chat
+-- @param Player isTeam @{Player} who speaks
```

Comment:
Should not be called `isTeam`, also wouldn't it rather be `... who chats` instead of `... who speaks`?

Reply:
all hooks used `isTeam`, I just used the same wording

yes

Reply:
This is still wrong, as this is the player who speaks and not the bool, which is described in the line below, so the variable name is wrong ;)

Reply:
oh, I see now, this is a typo! thanks